tools/cleanup_duplicate_host_guest_devices.rb

```diff
@@ -45,6 +46,22 @@
 
 return if opts[:dry_run]
 
-GuestDevice.destroy(guest_devices_to_delete) unless opts[:dry_run]
+associations = %i[has_one has_many].flat_map { |assoc| GuestDevice.reflect_on_all_associations(assoc) }
+dependents   = associations.select { |assoc| assoc.options[:dependent].present? }
+
+total_slices = (guest_devices_to_delete.count / opts[:page_size].to_f).ceil
+guest_devices_to_delete.each_slice(opts[:page_size]).with_index do |slice, index|
+  puts "Destroying slice #{index + 1} of #{total_slices}..."
+  GuestDevice.delete(slice)
+
+  dependents.each do |assoc|
+    delete_meth = assoc.options[:dependent]
+    foreign_key = assoc.join_keys.key
+    ids = assoc.klass.where(foreign_key => slice).pluck(:id)
```

Comment:
do we want `pluck` - not sure if `pluck` is better here or to just leave it as a `select`

also, `delete` and `pluck` may be good, but `destroy` would be better to just bring back the records